OsmAnd-java/src/main/java/net/osmand/binary/BinaryMapPoiReaderAdapter.java

```diff
@@ -740,6 +742,16 @@ private Amenity readPoiPoint(int left31, int right31, int top31, int bottom31,
 						am.setRoutePoint(arp);
 					}
 				}
+				if (hasLocation) {
+					if (precisionXY != 0) {
+						int[] xy = MapUtils.calculateFinalXYFromBaseAndPrecisionXY(24, 26, precisionXY, x >> 7, y >> 7, true);
+						int x31 = xy[0] << 5;
+						int y31 = xy[1] << 5;
+						am.setLocation(MapUtils.get31LatitudeY(y31), MapUtils.get31LongitudeX(x31));
+					} else {
+						am.setLocation(MapUtils.get31LatitudeY(y), MapUtils.get31LongitudeX(x));
+					}
+				}
 				return am;
 			case OsmandOdb.OsmAndPoiBoxDataAtom.DX_FIELD_NUMBER:
 				x = (codedIS.readSInt32() + (px << (24 - zoom))) << 7;
```

Comment:
shifting here to 7 and then later shifting back to 7 again, unnecessary operation

Reply:
Then need to do shift in if statement if (left31 > x || right31 < x || top31 > y || bottom31 < y) because x and y will be in 24 zoom format